database/rrdset.c

```diff
@@ -1381,13 +1381,29 @@ void rrdset_done(RRDSET *st) {
                     if(!(rrddim_flag_check(rd, RRDDIM_FLAG_DONT_DETECT_RESETS_OR_OVERFLOWS)))
                         storage_flags = SN_EXISTS_RESET;
 
-                    rd->last_collected_value = rd->collected_value;
-                }
+                    uint64_t last = (uint64_t)rd->last_collected_value;
+                    uint64_t new = (uint64_t)rd->collected_value;
+                    uint64_t max = (new > last) ? new : last;
```

Comment:
It will be better to store the maximum value in the database. The probability of changing a counter type on the fly is much less than the probability of sudden spikes.

Reply:
You are right.
Once we do this, the only way to change it would be to remove the dimension from disk.

Reply:
ok, did it per session. So, every time netdata starts, this is reset.